tools/cmake/cbmc_recipes.cmake

```diff
@@ -0,0 +1,323 @@
+    ${cbmc_proof_name}_0010_project.goto
+    PROPERTIES
+    INCLUDE_DIRECTORIES "${cbmc_compile_includes}"
+    COMPILE_DEFINITIONS "${cbmc_compile_definitions}"
+    COMPILE_OPTIONS     "${cbmc_compile_options}"
+)
+
+# Function removal
+#
+# If funs_to_remove is a non-empty list of functions, return a goto-instrument
+# command that strips those functions out of the binary.  If the list is empty,
+# return a command to copy the binary without changing it.
+function(cbmc_remove_functions funs_to_remove binary)
+    list(LENGTH ${funs_to_remove} n_funs_to_remove)
+
+    if(${n_funs_to_remove})
+      list(JOIN ${funs_to_remove} ";--remove-function-body;" cmd_list)
+      list(PREPEND
+          cmd_list
+          goto-instrument
+          --verbosity ${CBMC_VERBOSITY}
+          --remove-function-body
+      )
+      list(APPEND
+          cmd_list
+          ${cbmc_proof_name}_0010_${binary}.goto
+          ${cbmc_proof_name}_0020_${binary}_functions_removed.goto
+      )
+
+    else()
+      set(cmd_list
+          cmake -E copy
+          ${cbmc_proof_name}_0010_${binary}.goto
+          ${cbmc_proof_name}_0020_${binary}_functions_removed.goto
+      )
+    endif()
+
+    add_custom_command(
+        COMMENT "Removing ${n_funs_to_remove} functions from ${cbmc_proof_name}"
+        DEPENDS ${cbmc_proof_name}_0010_${binary}.goto
+        OUTPUT  ${cbmc_proof_name}_0020_${binary}_functions_removed.goto
+        COMMAND ${cmd_list}
+    )
+endfunction()
+
+cbmc_remove_functions(cbmc_harness_remove harness)
+cbmc_remove_functions(cbmc_project_remove project)
+
+add_custom_command(
+    COMMENT "Linking ${cbmc_proof_name}"
+    DEPENDS
+        ${cbmc_proof_name}_0020_harness_functions_removed.goto
+        ${cbmc_proof_name}_0020_project_functions_removed.goto
+    OUTPUT ${cbmc_proof_name}_0030_linked.goto
+    COMMAND
+        goto-cc --function harness
+        ${cbmc_proof_name}_0020_harness_functions_removed.goto
+        ${cbmc_proof_name}_0020_project_functions_removed.goto
+        -o ${cbmc_proof_name}_0030_linked.goto
+)
+
+add_custom_command(
+    COMMENT "Adding standard library to ${cbmc_proof_name}"
+    DEPENDS
+        ${cbmc_proof_name}_0030_linked.goto
+    OUTPUT
+        ${cbmc_proof_name}_0040_library_added.goto
+    COMMAND
+        goto-instrument --add-library --verbosity ${CBMC_VERBOSITY}
+        ${cbmc_proof_name}_0030_linked.goto
+        ${cbmc_proof_name}_0040_library_added.goto
+)
+
+add_custom_command(
+    COMMENT "Dropping unused functions from ${cbmc_proof_name}"
+    DEPENDS
+        ${cbmc_proof_name}_0040_library_added.goto
+    OUTPUT
+        ${cbmc_proof_name}_0050_unused_functions_dropped.goto
+    COMMAND
+        goto-instrument --drop-unused-functions --verbosity ${CBMC_VERBOSITY}
+        ${cbmc_proof_name}_0040_library_added.goto
+        ${cbmc_proof_name}_0050_unused_functions_dropped.goto
+)
+
+list(LENGTH cbmc_nondet_static_exclude cnse_length)
+if(${cbmc_nondet_static} OR ${cnse_length})
```

Comment:
Comment that you know exactly one is true given the sanity checking at top level of this file.